Mathlib/Analysis/Convex/Radon.lean

```diff
@@ -0,0 +1,190 @@
+/-
+Copyright (c) 2023 Vasily Nesterov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Vasily Nesterov
+-/
+import Mathlib.Analysis.Convex.Combination
+import Mathlib.LinearAlgebra.AffineSpace.Independent
+import Mathlib.Tactic.Linarith
+
+/-!
+# Radon's convexity theorem
+
+Radon's theorem on convex sets states that any affine dependent set can be partitioned into two sets
+whose convex hulls intersect.
+
+## Main results
+
+* `Radon_partition`: Radon convexity theorem
+
+## Tags
+convex hull, radon
+
+-/
+
+
+open Set Finset
+open BigOperators
+
+universe u
+
+variable {𝕜 : Type*} {E : Type u} [LinearOrderedField 𝕜] [AddCommGroup E] [Module 𝕜 E]
+
+/-- If `s` is a set of affine dependent vectors, it has subset `I` with the property that
+convex hulls of `I` and `s \ I` intersect. -/
+theorem Radon_partition (s : Set E)
+(h : ¬AffineIndependent 𝕜 ((↑) : s → E)) : ∃ (I : Set E), (I ⊆ s) ∧
+(Set.Nonempty ((convexHull 𝕜 I) ∩ (convexHull 𝕜 (s \ I)))) := by
```

Comment:
```suggestion
    (h : ¬AffineIndependent 𝕜 ((↑) : s → E)) : ∃ (I : Set E), (I ⊆ s) ∧
    (Set.Nonempty ((convexHull 𝕜 I) ∩ (convexHull 𝕜 (s \ I)))) := by
```
I don't think we have a documentation for this yet, but the convention in Lean4 is that everything before the `:=` should be indented twice.